# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/WebIdentityParent.cpp
# Commit: 94b9d6ff8926
# Full Hash: 94b9d6ff8926b6b566916a4bc06a4d4718afe209
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-08-26 09:53:22
# Description:
#   Bug 1983101 - Fix crash in navigator.login.setStatus - r=emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D262181
# ==============================================================================

diff -r ebb71534c42a -r 94b9d6ff8926 dom/credentialmanagement/identity/WebIdentityParent.cpp
--- a/dom/credentialmanagement/identity/WebIdentityParent.cpp	Mon Aug 25 18:57:11 2025 +0000
+++ b/dom/credentialmanagement/identity/WebIdentityParent.cpp	Mon Aug 25 19:04:15 2025 +0000
@@ -91,6 +91,7 @@
   WindowGlobalParent* manager = static_cast<WindowGlobalParent*>(Manager());
   if (!manager) {
     aResolver(NS_ERROR_FAILURE);
+    return IPC_OK();
   }
   nsIPrincipal* principal = manager->DocumentPrincipal();
   if (!principal) {